src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -0,0 +1,285 @@
+﻿using ApacheOrcDotNet.Encodings;
+using ApacheOrcDotNet.OptimizedReader.Encodings;
+using ApacheOrcDotNet.OptimizedReader.Infrastructure;
+using ApacheOrcDotNet.Protocol;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace ApacheOrcDotNet.OptimizedReader.Buffers
+{
+    [SkipLocalsInit]
+    public abstract class BaseColumnBuffer<TOutput>
+    {
+        private readonly long[] _numericStreamBuffer;
```

Comment:
These are intermediary buffers that are only used in the base class. So it made sense to keep them local.

Reply:
Let me look at that again.  To me having buffers whose type is determined by the superclass hang around just in case a particular superclass wants to use them makes less sense than just including them in the superclass.